src/main/java/by/itech/lab/supplier/service/impl/WarehouseServiceImpl.java

```diff
@@ -1,42 +1,79 @@
 package by.itech.lab.supplier.service.impl;
 
+import by.itech.lab.supplier.domain.Address;
+import by.itech.lab.supplier.domain.Customer;
+import by.itech.lab.supplier.domain.Warehouse;
 import by.itech.lab.supplier.dto.WarehouseDto;
+import by.itech.lab.supplier.dto.mapper.AddressMapper;
+import by.itech.lab.supplier.dto.mapper.UserMapper;
 import by.itech.lab.supplier.dto.mapper.WarehouseMapper;
 import by.itech.lab.supplier.exception.ResourceNotFoundException;
+import by.itech.lab.supplier.repository.AddressRepository;
+import by.itech.lab.supplier.repository.CustomerRepository;
+import by.itech.lab.supplier.repository.UserRepository;
 import by.itech.lab.supplier.repository.WarehouseRepository;
 import by.itech.lab.supplier.service.WarehouseService;
 import lombok.AllArgsConstructor;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Optional;
 
 @Service
 @AllArgsConstructor
 public class WarehouseServiceImpl implements WarehouseService {
 
     private final WarehouseRepository warehouseRepository;
+    private final CustomerRepository customerRepository;
+    private final AddressRepository addressRepository;
+    private final UserRepository userRepository;
     private final WarehouseMapper warehouseMapper;
+    private final AddressMapper addressMapper;
```

Comment:
if the address has its own mapper - maybe it should have its own service?

Reply:
we don't need service class for Address, cuz this information write and update through warehouse only

Reply:
in any case it has its own repo, mapper, entity, dto, if you going to manage addresses in the scope of the warehouse - you dont need a repo and changes must be applied together with the warehouse entity